mavutil.py

```diff
@@ -2003,7 +1982,48 @@ def mode_string_v09(msg):
     9: 'SURFACE',
     16: 'POSHOLD',
     19: 'MANUAL',
-    }
+}
+
+AP_MAV_TYPE_MODE_MAP_DEFAULT = {
+    # copter
+    mavlink.MAV_TYPE_HELICOPTER:  mode_mapping_acm,
+    mavlink.MAV_TYPE_TRICOPTER:   mode_mapping_acm,
+    mavlink.MAV_TYPE_QUADROTOR:   mode_mapping_acm,
+    mavlink.MAV_TYPE_HEXAROTOR:   mode_mapping_acm,
+    mavlink.MAV_TYPE_OCTOROTOR:   mode_mapping_acm,
+    mavlink.MAV_TYPE_DECAROTOR:   mode_mapping_acm,
+    mavlink.MAV_TYPE_DODECAROTOR: mode_mapping_acm,
+    mavlink.MAV_TYPE_COAXIAL:     mode_mapping_acm,
+    # plane
+    mavlink.MAV_TYPE_FIXED_WING: mode_mapping_apm,
+    # rover
+    mavlink.MAV_TYPE_GROUND_ROVER: mode_mapping_rover,
+    # boat
+    mavlink.MAV_TYPE_SURFACE_BOAT: mode_mapping_rover, # for the time being
+    # tracker
+    mavlink.MAV_TYPE_ANTENNA_TRACKER: mode_mapping_tracker,
+    # sub
+    mavlink.MAV_TYPE_SUBMARINE: mode_mapping_sub,
+}
+
+
+try:
+    # Allow for using custom mode maps by importing a dict named
+    # AP_MAV_TYPE_MODE_MAP_CUSTOM from "~/.pymavlink/ap_custom_mode_map.py"
+    # and using it to extend the hard-coded AP_MAV_TYPE_MODE_MAP_DEFAULT dict.
+    import sys
+    from os.path import expanduser
+
+    sys.path.append(expanduser('~') + '/.pymavlink')
```

Comment:
if doing this then the sys.path.append needs to be conditional on the existance of ~/.pymavlink/ap_custom_mode_map as otherwise it polluting the system path for all modules
but I'd much this was a data file, such as json format, in ~/.pymavlink/modemap.json, and it parses that file to replace (or augment?) the dictionary. I think that would be more portable and less likely to break things